api/lib/resolvers.js

```diff
@@ -47,77 +46,21 @@ function localStore(dataSources, networkId) {
   }
 }
 
-// TODO updating the validators with the profiles should happen already in the store so we don't query the db all the time
-async function getValidatorInfosMap(store) {
-  const validatorInfo = await store.db.getValidatorsInfo()
-  const validatorInfoMap = keyBy(validatorInfo, 'operator_address')
-  return validatorInfoMap
-}
-
-function enrichValidator(validatorInfo, validator) {
-  const picture = validatorInfo ? validatorInfo.picture : undefined
-  const name =
-    validatorInfo && validatorInfo.name
-      ? validatorInfo.name
-      : validator.name || formatBech32Reducer(validator.operatorAddress)
-
-  return {
-    ...validator,
-    name,
-    picture: picture === 'null' || picture === 'undefined' ? undefined : picture
-  }
-}
-
-async function addPopularityToValidators(validators, dataSources, networkId) {
-  return Promise.all(
-    validators.map(async (validator) => {
-      // popularity is actually the number of views of a validator on their page
-      const popularity = await localStore(
-        dataSources,
-        networkId
-      ).db.getValidatorViews(validator.operatorAddress, networkId)
-      // we add the popularity field to the validator
-      return {
-        ...validator,
-        popularity: popularity || 0
-      }
-    })
-  )
-}
-
 async function validators(
   _,
   { networkId, searchTerm, activeOnly, popularSort },
   { dataSources }
 ) {
   await localStore(dataSources, networkId).dataReady
   let validators = Object.values(localStore(dataSources, networkId).validators)
-  validators = await addPopularityToValidators(
-    validators,
-    dataSources,
-    networkId
-  )
-  function compare(a, b) {
-    let comparison = 0
-    if (a.popularity < b.popularity) {
-      comparison = 1
-    } else if (a.popularity > b.popularity) {
-      comparison = -1
-    }
-    return comparison
-  }
-  // we always sort validators by popularity
-  validators.sort(compare)
   if (activeOnly) {
     validators = validators.filter(({ status }) => status === 'ACTIVE')
   }
-  const validatorInfoMap = await getValidatorInfosMap(
-    localStore(dataSources, networkId)
-  )
-  validators = validators.map((validator) =>
-    enrichValidator(validatorInfoMap[validator.operatorAddress], validator)
-  )
-
```

Comment:
Uuuh nice! So we do all this ordering and filtering in store now

Reply:
enriching is happening once instead of every time

Reply:
Lovely!!